packages/cli/src/oclif/commands/scaffold.js

```diff
@@ -0,0 +1,213 @@
+const _ = require('lodash');
+const path = require('path');
+const colors = require('colors/safe');
+
+const { flags } = require('@oclif/command');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const { buildFlags } = require('../buildFlags');
+const {
+  ensureDir,
+  fileExistsSync,
+  readFile,
+  writeFile
+} = require('../../utils/files');
+const { camelCase, snakeCase } = require('../../utils/misc');
+const { splitFileFromPath } = require('../../utils/string');
+
+// what is the `resources: {}` app definition point?
+const typeMap = {
+  resource: 'resources',
+  trigger: 'triggers',
+  search: 'searches',
+  create: 'creates'
+};
+
+function createTemplateContext(type, name) {
+  const contextKey = snakeCase(name);
+
+  // where will we create/required the new file?
+  const destMap = {
+    resource: `resources/${contextKey}`,
+    trigger: `triggers/${contextKey}`,
+    search: `searches/${contextKey}`,
+    create: `creates/${contextKey}`
+  };
+
+  return {
+    CAMEL: camelCase(name),
+    KEY: contextKey,
+    NOUN: _.capitalize(name),
+    LOWER_NOUN: name.toLowerCase(),
+    INPUT_FIELDS: '',
+    TYPE: type,
+    TYPE_PLURAL: type === 'search' ? `${type}es` : `${type}s`,
+    // resources need an extra line for tests to "just run"
+    MAYBE_RESOURCE: type === 'resource' ? 'list.' : '',
+    defaultDest: destMap[type]
+  };
+}
+
+function getTemplatePath(type) {
+  return path.join(__dirname, '../../..', `scaffold/${type}.template.js`);
+}
+
+class ScaffoldCommand extends BaseCommand {
+  async writeTemplateFile(type, templateContext, dest) {
+    const templatePath = getTemplatePath(type);
+    const destPath = path.join(process.cwd(), `${dest}.js`);
+    const preventOverwrite = !this.flags.force;
+
+    if (preventOverwrite && fileExistsSync(destPath)) {
+      const [filename, location] = splitFileFromPath(destPath);
+
+      return this.error(
```

Comment:
minor, but you don't need to `return` here. This'll throw as normal (I _think_; if you're seeing different behavior, feel free to ignore)

Reply:
Force of habit to always return something—and also needing to read up on oclif's API 😅

Reply:
haha yeah. [this](https://oclif.io/docs/commands) is a good starting point, plus [the TS reference](https://github.com/oclif/command/blob/master/src/command.ts).

Reply:
I think I'd like it better if it was `this.throwError`. 

If I'm scanning the code and see `this.error` I don't automatically read that we break out of that block.